delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/routes/FilesRoutes.scala

```diff
@@ -88,17 +92,21 @@ final class FilesRoutes(
                     entity(as[LinkFile]) { case LinkFile(filename, mediaType, path) =>
                       emit(
                         Created,
-                        files.createLink(storage, ref, filename, mediaType, path, tag).tapEval(indexUIO(ref, _, mode))
+                        files.createLink(storage, ref, filename, mediaType, path, tag).index(mode).attemptN
                       )
                     },
                     // Create a file without id segment
                     extractRequestEntity { entity =>
-                      emit(Created, files.create(storage, ref, entity, tag).tapEval(indexUIO(ref, _, mode)))
+                      emit(
+                        Created,
+                        files.create(storage, ref, entity, tag).index(mode).attemptN
```

Comment:
I prefer the explicit `attemptNarrow` here. Alternatively, in some routes we have more narrow emit methods which hide this "boilerplate"

Reply:
Possibly a better name would make it work..?

Reply:
Yeah I was just messing around to see what it'd look like - not worth it really. Put it back